test/data/test_dataset.py

```diff
@@ -100,19 +101,23 @@ def test_converter():
     assert len(tf) == len(dataset)
 
 
-def test_multi_categorical_materialization():
-    data = {'a': ['A,B', 'B,C,A', '', 'B', None]}
+def test_multicategorical_materialization():
+    data = {'multicat_col': ['A|B', 'B|C|A', '', '', 'B', 'B|A', None]}
     df = pd.DataFrame(data)
-    dataset = Dataset(df, {'a': stype.multi_categorical})
+    dataset = Dataset(df, {'multicat_col': stype.multicategorical},
+                      col_to_sep={'multicat_col': '|'})
     dataset.materialize()
-    feat = dataset.tensor_frame.feat_dict[stype.multi_categorical]
-    assert torch.equal(feat[0, 0], torch.tensor([1, 0], device=feat.device))
-    assert torch.equal(feat[1, 0], torch.tensor([0, 2, 1], device=feat.device))
-    assert feat[2, 0].numel() == 0
-    assert torch.equal(feat[4, 0], torch.tensor([-1], device=feat.device))
-    assert StatType.COUNT in dataset.col_stats['a']
-    assert dataset.col_stats['a'][StatType.COUNT][0] == ['B', 'A', 'C']
-    assert dataset.col_stats['a'][StatType.COUNT][1] == [3, 2, 1]
+    feat = dataset.tensor_frame.feat_dict[stype.multicategorical]
+    assert torch.equal(feat[0, 0], torch.tensor([1, 0]))
+    assert torch.equal(feat[1, 0], torch.tensor([0, 2, 1]))
+    assert torch.equal(feat[6, 0], torch.tensor([-1]))
```

Comment:
Can we add tests for 2, 3, 4, 5 as well?

Reply:
While I was adding tests, I realized that I didn't account for the case where some categories can appear multiple times. I added some code and fixed test cases to accommodate that case.

Reply:
Good catch. thanks